tests/unit/Internal/Primitives/CallableTypeTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Facile\PhpCodec\Internal\Primitives;
+
+use Eris\TestTrait;
+use Facile\PhpCodec\Codecs;
+use Tests\Facile\PhpCodec\BaseTestCase;
+use Tests\Facile\PhpCodec\GeneratorUtils;
+
+class CallableTypeTest extends BaseTestCase
+{
+    use TestTrait;
+
+    public function testLaws(): void
+    {
+        $this
+            ->forAll(
+                GeneratorUtils::scalar(),
+                static function (): void {
```

Comment:
Fine, but it is a constant. Try to use a callable generator like the one above to provide other values.